acceptance_tests/features/steps/add_a_survey.py

```diff
@@ -0,0 +1,67 @@
+from behave import given, when, then
+
+from acceptance_tests import browser
+from controllers.database_controller import select_iac
+from acceptance_tests.features.pages import surveys_todo, add_survey
+
+
+@when('access to do list in my surveys')
+def surveys_to_do_list(context):
+    surveys_todo.go_to()
+
+
+@given('selects to add a new survey')
+@when('they add a new survey')
+@then('they are able to add a new survey')
+def add_a_new_survey(context):
+    browser.find_by_id('ADD_SURVEY_BTN').click()
+
+
+@then('they are able to enter an enrolment code')
+def enter_enrolment_code(context):
+    browser.find_by_id('ENROLMENT_CODE_FIELD')
+
+
+@given('the user has entered a valid enrolment code')
+@when('they enter a valid enrolment code')
+def enter_valid_enrolment_code(context):
+    add_survey.go_to()
+    enrolment_code = select_iac()
+    browser.driver.find_element_by_id('ENROLEMENT_CODE_FIELD').send_keys(enrolment_code)
+    browser.find_by_id('CONTINUE_BTN').click()
+
+
+@then('they are to be presented with the survey and organisation that they are enrolling for')
+def view_confirmation_page(context):
+    browser.find_by_id('confirm-org-title')
+
+
+@when('they enter an invalid enrolment code')
+def enter_invalid_enrolment_code(context):
+    browser.driver.find_element_by_id('ENROLEMENT_CODE_FIELD').send_keys('222thu78nj7m')
+    browser.find_by_id('CONTINUE_BTN').click()
+
+
+@then('they are to be notified')
+def invalid_enrolment_code_notification(context):
+    browser.find_by_id('error_notification')
+
+
+@when('they continue and confirm that the organisation and survey that they are enrolling for is correct')
+def confirm_organisation_and_continue(context):
+    browser.find_by_id('CONFIRM_SURVEY_BTN').click()
+
+
+@then('the new survey is to be listed in My Surveys and confirmation is presented to the user')
+def confirmation_presented_for_new_survey(context):
+    browser.find_by_id('NEW_SURVEY_NOTIF')
```

Comment:
I'm not a fan of shortening stuff like this, I think having the full word is better `NEW_SURVEY_NOTIFCATION` or `NEW_SURVEY_NOTIFY`... Although this is just preference.

Reply:
I do agree. However this is the name given in frontstage so without changing there it needs to be this way in the tests.

Reply:
Ah this was me 😅 I just gave them a quick name when I was building these pages. It's probably better in hindsight to give it the full name, but we can change that in the future.